### ISSUE TITLE
Alphanumeric sort of lists

### ISSUE THREAD

**Issue by wyardley** [terraform-linters/tflint#1811]
## Introduction
Similar to #783, but for lists / object keys

## Proposal
When a data structure like a list or object is used, we often like to have the elements sorted alphanumerically. It would be nice to have an optional (and _not_ enabled by default) linter rule to enforce this. Open to suggestions about, for example, if all unquoted values are first or last, but overall, should work like a naïve alphnumeric sort that factors in numeric values

For example:

passes
```hcl
  bar = {
    groups = [
      var.bar.email,
      var.foo.email,
      "xyz",
    ]
    service_accounts = [data.terraform_remote_state.foo.outputs.xyz, "foo"]
  }
  foo = [0, 1, 2, 3, 100, "foo", "bar"]
```

generates errors
```hcl
  bar = {
    groups = [
      var.foo.email,
      "xyz",
      var.bar.email,
    ]
    service_accounts = ["foo", data.terraform_remote_state.foo.outputs.xyz]
  }
  foo = [0, 3, 1, 100, 2, "bar", "foo"]
```
